packages/webapp/src/routes/__tests__/signin/callback.spec.js

```diff
@@ -66,148 +67,107 @@ describe('Signin callback handler', () => {
       }
     })
   })
-  it('Should redirect to the allocations task list when an authenticated allocation user has made no applications', done => {
-    jest.isolateModules(async () => {
-      try {
-        jest.resetAllMocks()
-        jest.mock('../../../utils/http.js')
-        const http = require('../../../utils/http.js')
-        http.postJson = jest.fn().mockImplementation(() => {
-          return []
-        })
-        auth.authenticate = jest.fn().mockImplementation(() => {
-          return {
-            idTokenClaims: {
-              contactId: 'mock contact id'
-            }
-          }
-        })
 
-        const session = new Session()
-        session.set(constants.redisKeys.PRE_AUTHENTICATION_ROUTE, '/developer/mock-route')
+  it('should redirect to the registrations tasklist when an authenticated registration user has one associated applications', done => {
+    const mockApplications = [{
+      applicationStatus: 'IN PROGRESS',
+      applicationReference: 'mock-ref'
+    }]
+    const preAuthenticationRoute = '/land/mock-route'
+    const redirectUrl = constants.routes.REGISTER_LAND_TASK_LIST
+    processRedirectionByApplicationType(mockApplications, preAuthenticationRoute, redirectUrl, done)
+  })
 
-        let redirectArgs = ''
-        const h = {
-          redirect: (...args) => {
-            redirectArgs = args
-          }
-        }
-        const getHandler = callback[0].handler
-        await getHandler({ yar: session, query: {} }, h)
-        expect(redirectArgs[0]).toEqual(constants.routes.DEVELOPER_TASKLIST)
-        expect(auth.authenticate).toHaveBeenCalledTimes(1)
-        done()
-      } catch (err) {
-        done(err)
-      }
-    })
+  it('should redirect to the registrations tasklist when an authenticated registration user has more than one associated applications', done => {
```

Comment:
Update message registrations tasklist is incorrect, should be biodiversity gain sites.

Reply:
or registrations dashboard